src/FakeItEasy/Core/ArgumentValueFormatter.cs

```diff
@@ -37,16 +34,26 @@ public virtual string GetArgumentValueAsString(object? argumentValue)
 
             var argumentType = argumentValue.GetType();
 
-            var formatter = this.cachedFormatters.GetOrAdd(argumentType, this.ResolveTypeFormatter);
-
-            try
-            {
-                return formatter.GetArgumentValueAsString(argumentValue);
-            }
-            catch (Exception ex) when (formatter.GetType().GetTypeInfo().Assembly != typeof(ArgumentValueFormatter).GetTypeInfo().Assembly)
+            foreach (var formatter in this.GetTypeFormatterCandidates(argumentType))
             {
-                throw new UserCallbackException(ExceptionMessages.UserCallbackThrewAnException($"Custom argument value formatter '{formatter}'"), ex);
+                try
+                {
+                    var formattedValue = formatter.GetArgumentValueAsString(argumentValue);
+                    if (formattedValue is object)
+                    {
+                        return formattedValue;
+                    }
+                }
+                catch when (formatter.GetType().GetTypeInfo().Assembly != typeof(ArgumentValueFormatter).GetTypeInfo().Assembly)

```

Comment:
I think there are several places where we do this kind of check... Maybe we should have a `IsFakeItEasyType` method somewhere. (out of scope for this PR, of course)

Reply:
I cannot find another place!
We _do_ save away the the assembly in `TypeCatalogue` though…